diff --git a/src/closures.c b/src/closures.c
index 02cf78f..0331fc5 100644
--- a/src/closures.c
+++ b/src/closures.c
@@ -418,13 +418,15 @@ ffi_closure_free (void *ptr)
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <fcntl.h>
+#if !defined(__MINGW32CE__)
 #include <errno.h>
+#endif
 #ifndef _MSC_VER
 #include <unistd.h>
 #endif
 #include <string.h>
 #include <stdio.h>
-#if !defined(_WIN32)
+#if !defined(_WIN32) && !defined(__MINGW32CE__)
 #ifdef HAVE_MNTENT
 #include <mntent.h>
 #endif /* HAVE_MNTENT */
diff --git a/src/mips/ffi.c b/src/mips/ffi.c
index 5c8c6bc..47aec6f 100644
--- a/src/mips/ffi.c
+++ b/src/mips/ffi.c
@@ -33,8 +33,12 @@
 #include <stdlib.h>
 #include <stdio.h>
 
+#if defined(__MINGW32CE__)
+#include <windows.h>
+#endif
+
 #ifdef __GNUC__
-#  if (__GNUC__ > 4) || ((__GNUC__ == 4) && (__GNUC_MINOR__ >= 3))
+#  if defined(__clang__) || (__GNUC__ > 4) || ((__GNUC__ == 4) && (__GNUC_MINOR__ >= 3))
 #    define USE__BUILTIN___CLEAR_CACHE 1
 #  endif
 #endif
@@ -928,7 +932,10 @@ ffi_prep_closure_loc (ffi_closure *closure,
   closure->user_data = user_data;
 
 #if !defined(__FreeBSD__)
-#ifdef USE__BUILTIN___CLEAR_CACHE
+#if defined(__MINGW32CE__)
+  FlushInstructionCache(GetCurrentProcess(), clear_location,
+                        FFI_TRAMPOLINE_SIZE);
+#elif defined(USE__BUILTIN___CLEAR_CACHE)
   __builtin___clear_cache(clear_location, clear_location + FFI_TRAMPOLINE_SIZE);
 #else
   cacheflush (clear_location, FFI_TRAMPOLINE_SIZE, ICACHE);
diff --git a/src/mips/ffitarget.h b/src/mips/ffitarget.h
index 7ec8daa..72d1aeb 100644
--- a/src/mips/ffitarget.h
+++ b/src/mips/ffitarget.h
@@ -39,6 +39,10 @@
 #define _MIPS_SIM_ABI32		1
 #define _MIPS_SIM_NABI32	2
 #define _MIPS_SIM_ABI64		3
+#elif defined(__MINGW32CE__)
+#define _MIPS_SIM_ABI32 1
+#define _MIPS_SIM_NABI32 2
+#define _MIPS_SIM_ABI64 3
 #elif !defined(__OpenBSD__) && !defined(__FreeBSD__) && !defined(__linux__)
 # include <sgidefs.h>
 #endif
diff --git a/src/mips/o32.S b/src/mips/o32.S
index 03ceed5..07d09b7 100644
--- a/src/mips/o32.S
+++ b/src/mips/o32.S
@@ -453,6 +453,7 @@ $LFE2:
 
 /* DWARF-2 unwind info. */
 
+#if !defined(__MINGW32CE__)
 	.section	.eh_frame,"a",@progbits
 $Lframe0:
 	.4byte	$LECIE0-$LSCIE0	 # Length of Common Information Entry
@@ -559,6 +560,7 @@ $LASFDE2:
 $LEFDE2:
 
 #endif
+#endif /* !defined(__MINGW32CE__) */
 
 #if defined __ELF__ && defined __linux__
 	.section .note.GNU-stack,"",%progbits
